nilearn/plotting/tests/test_img_plotting/test_plot_carpet.py

```diff
@@ -46,12 +51,10 @@ def test_plot_carpet(img_4d_mni, img_3d_ones_mni, img_4d_long_mni):
         img_4d_long_mni.shape[-1] / 2
     )
     assert plotted_array.size == n_items
-    plt.close(display)
 
 
-def test_plot_carpet_with_atlas(img_4d_mni, img_atlas):
+def test_plot_carpet_with_atlas(pyplot, img_4d_mni, img_atlas):
```

Comment:
Sorry, what is the point ?

Reply:
This uses the same fixture as in sklearn: 

https://scikit-learn.org/dev/developers/contributing.html#testing-and-improving-test-coverage

![image](https://github.com/user-attachments/assets/19683b38-7dd9-4a10-8a19-ceafcc835474)

The fixture is in nilearn/plotting/tests/conftest.py

- checks if matplotlib is there, skips test if not
- setup: close any open figure before the test is run
- teardown: closes any figure open by the test

hence why I am removing all the plt.close('all') calls: the fixture does it for us even if the test fails: this makes sure tests are more independent and do not pollute each other

Reply:
probably worth adding something similar in our own doc